[BACKPORT-v1.14] random: Add syscalls for random subsystem

ceolin opened the pull request:
Create syscalls to make possible using random APIs from user mode
threads. These APIs can have different implementations, like using
entropy driver or Xoroshiro128. Some of these implementations also
have some globals to preserve state between calls.

Make it run entire in user space would require user adding these
globals to their memeory domains and/or grant access to entropy device.
Syscalls simplify its usage.

Fixes #26499
Backport of #25980

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
random: Add syscalls for random subsystem

Create syscalls to make possible using random APIs from user mode
threads. These APIs can have different implementations, like using
entropy driver or Xoroshiro128. Some of these implementations also
have some globals to preserve state between calls.

Make it run entire in user space would require user adding these
globals to their memeory domains and/or grant access to entropy device.
Syscalls simplify its usage.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

andrewboie commented:
I'm really getting confused by distinctions between this PR, #25980, and #25890

ceolin commented:
> I'm really getting confused by distinctions between this PR, #25980, and #25890

this one is basically #25980 backported to 1.14. Forget about #25890, I will close that, it was an attempt to have random subsys in usermode without syscalls.

andrewboie commented:
> 
> 
> > I'm really getting confused by distinctions between this PR, #25980, and #25890
> 
> this one is basically #25980 backported to 1.14. Forget about #25890, I will close that, it was an attempt to have random subsys in usermode without syscalls.

Gotcha thanks!!

Commit:
random: Include header where it is used

Unit tests were failing to build because random header was
included by kernel_includes.h. The problem is that rand32.h
includes a generated file that is either not generated or not
included when building unit tests. Also, it is better to limit
the scope of this file to where it is used.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

nashif commented:
@ceolin  do we have a bug associated with this?

ceolin commented:
> @ceolin do we have a bug associated with this?

nope, we don't have an issue associated.

nashif commented:
> nope, we don't have an issue associated.

then you need to create one and document what the problem is. see https://github.com/zephyrproject-rtos/zephyr/wiki/Zephyr-Change-Control-Board-%28CCB%29

ceolin commented:
> > nope, we don't have an issue associated.
> 
> then you need to create one and document what the problem is. see [zephyrproject-rtos/zephyr/wiki/Zephyr-Change-Control-Board-%28CCB%29](https://github.com/zephyrproject-rtos/zephyr/wiki/Zephyr-Change-Control-Board-%28CCB%29)

#26499
